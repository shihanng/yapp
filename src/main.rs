use zellij_tile::prelude::*;

use std::collections::BTreeMap;

#[derive(Default)]
struct State {
    // the state of the plugin
}

register_plugin!(State);

// NOTE: you can start a development environment inside Zellij by running `zellij -l zellij.kdl` in
// this plugin's folder
//
// More info on plugins: https://zellij.dev/documentation/plugins

impl ZellijPlugin for State {
    fn load(&mut self, configuration: BTreeMap<String, String>) {
        // runs once on plugin load, provides the configuration with which this plugin was loaded
        // (if any)
        //
        // this is a good place to `subscribe` (https://docs.rs/zellij-tile/latest/zellij_tile/shim/fn.subscribe.html)
        // to `Event`s (https://docs.rs/zellij-tile/latest/zellij_tile/prelude/enum.Event.html)
        // and `request_permissions` (https://docs.rs/zellij-tile/latest/zellij_tile/shim/fn.request_permission.html)
    }
    fn update(&mut self, event: Event) -> bool {
        let mut should_render = false;
        // react to `Event`s that have been subscribed to (and the plugin has permissions for)
        // return true if this plugin's `render` function should be called for the plugin to render
        // itself
        should_render
    }
    fn pipe (&mut self, pipe_message: PipeMessage) -> bool {
        let mut should_render = false;
        // react to data piped to this plugin from the CLI, a keybinding or another plugin
        // read more about pipes: https://zellij.dev/documentation/plugin-pipes
        // return true if this plugin's `render` function should be called for the plugin to render
        // itself
        should_render
    }
    fn render(&mut self, rows: usize, cols: usize) {
        println!("Hi there! I have {rows} rows and {cols} columns");
    }
}
