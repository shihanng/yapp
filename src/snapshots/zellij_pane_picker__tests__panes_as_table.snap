---
source: src/main.rs
expression: "format!(\"\\u{1b}Pztable;{}\", state.panes_as_table(20).serialize())"
---
Pztable;4;4;32;84,97,98,32,32;32,73,68;80,97,110,101,32,84,105,116,108,101;0$32;84,97,98,32,49;32,32,49;80,97,110,101,32,49;x0$42;x84,97,98,32,49;x32,32,50;x80,97,110,101,32,50;0$32;84,97,98,32,50;32,53,53;80,97,110,101,32,53,53\
